### PR TITLE
docs: explanation

### PR DESCRIPTION
**Pull request by yanksyoon**
This PR implements explanation documentation for wordpress-k8s charm using the [Diataxis framework](https://diataxis.fr/explanation/).

Out of scope of this PR: other parts of documentation(tutorial, how-to guide). Later PRs will address the remaining documentations.

### CODE REVIEW

File: docs/explanation/lifecycle-events.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+current state and the desired state with new relation data to synchronize the replication
+instances. The reconciliation process is divided into 3 distinct steps: core, theme and plugin
+reconciliation. Core reconciliation setups up the necessary WordPress application configuration:
+secrets and database connection. Theme and Plugin respectively reconcile between currently
+installed themes and plugins with the incoming list of themes and plugins.
+
+### upgrade-charm
+
+Upgrade charm event is fired on the upgrade charm command `juju refresh wordpress-k8s`. Sets up new
+secrets for upgraded deployment of WordPress.
```
[the remark this answers is not in the document]
Comment (yanksyoon) [reply docs/explanation/lifecycle-events.md]: Fixed the description to "Sets up secrets in peer-relation databag for upgraded deployment of WordPress if not already set."
Thanks for the clarification!

File: docs/explanation/lifecycle-events.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+This event marks the charm’s storage availability. The name of the event derived from the name of
+the storage noted in the `metadata.yaml` configuration under "storage" key.
+`containers.wordpress.mounts.storage` and `storage.uploads` section. The storage filesystem maps to
+`/var/www/html/wp-content/uploads` directory of the WordPress application, which is used to store
+uploaded content from the WordPress user.
+
+### leader_elected
+
+This event is fired when juju elects a leader unit among the replica peers. Wordpress-k8s charm
+then responds by setting up secrets and sharing them with peers through peer relation databag.
```
[the remark this answers is not in the document]
Comment (yanksyoon) [reply docs/explanation/lifecycle-events.md]: Also added the description to include "if not already set" to clarify that it will only be generated once.
Thank you for the comment!

File: docs/explanation/oci-image.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+plugins and themes can be found at the reference section of the
+[documentation](https://charmhub.io/wordpress-k8s/docs/reference?channel=edge). Since WordPress is
+an application running on php, required libraries and dependencies are installed during the build
+process.
+
+WordPress application installation is done at runtime once during charm initialisation.
+To facilitate the WordPress installation process, WordPress cli is embedded in the OCI image during
+build step. The latest cli php archive file from source is used.
+
+Currently, WordPress version 5.9.3 is used alongside Ubuntu 20.04 base image. The Ubuntu base image
```
[the remark this answers is not in the document]
Comment (yanksyoon) [reply docs/explanation/oci-image.md:17]: Sounds like a good idea! Added :)

File: docs/explanation/oci-image.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+### wordpress-image
+
+The wordpress-image is custom built to include a default set of plugins and themes. The list of
+plugins and themes can be found at the reference section of the
+[documentation](https://charmhub.io/wordpress-k8s/docs/reference?channel=edge). Since WordPress is
+an application running on php, required libraries and dependencies are installed during the build
+process.
+
+WordPress application installation is done at runtime once during charm initialisation.
```
[the remark this answers is not in the document]
Comment (yanksyoon) [reply docs/explanation/oci-image.md]: Added the description to clarify that it is done during database setup.
```
WordPress application installation is done at runtime during database connection setup. This can
happen during database relation changed, database relation joined or database config changed
events.
```
Thank you for the clarification!

File: docs/explanation/lifecycle-events.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+### leader_elected
+
+This event is fired when juju elects a leader unit among the replica peers. Wordpress-k8s charm
+then responds by setting up secrets and sharing them with peers through peer relation databag if
+not already set.
+
+### config-changed
+
+WordPress-k8s charm reacts to any configuration change and runs reconciliation between the current
+state and the desired state. See the list of configurations in the reference documentation.
```
Comment (jdkandersson) [inline docs/explanation/lifecycle-events.md:31]: Should this include a link to the configuration page?

File: docs/explanation/oci-image.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+process.
+
+WordPress application installation is done at runtime during database connection setup. This can
+happen during database relation changed, database relation joined or database config changed
+events.
+To facilitate the WordPress installation process, WordPress cli is embedded in the OCI image during
+build step. The latest cli php archive file from source is used.
+
+Currently, WordPress version 5.9.3 is used alongside Ubuntu 20.04 base image. The Ubuntu base image
+is not being upgraded to 22.04 due to an unsupported php version 8 for
```
Comment (jdkandersson) [inline docs/explanation/oci-image.md:18]: This reads like we will never upgrade, perhaps reword it to say that it hasn't been upgraded yet due to this reason?

**Review by jdkandersson — DISMISSED** [review]

File: docs/explanation/lifecycle-events.md (hunk 5 above)
Comment (yanksyoon) [reply docs/explanation/lifecycle-events.md]: Added link to configurations tab! Thanks for the comment :)

**Review by jdkandersson — APPROVED** [review]

**Review by weiiwang01 — APPROVED** [review]
